server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -736,8 +717,47 @@ public VolumeVO allocVolume(CreateVolumeCmd cmd) throws ResourceAllocationExcept
 
         String userSpecifiedName = getVolumeNameFromCommand(cmd);
 
-        return commitVolume(cmd, caller, owner, displayVolume, zoneId, diskOfferingId, provisioningType, size, minIops, maxIops, parentVolume, userSpecifiedName,
+        VolumeVO volume= commitVolume(cmd, caller, owner, displayVolume, zoneId, diskOfferingId, provisioningType, size,
+                minIops,
+                maxIops, parentVolume, userSpecifiedName,
                 _uuidMgr.generateUuid(Volume.class, cmd.getCustomId()));
+
```

Comment:
@kioie can you modularise this code somewhat? as it has moved it is hard to compare with the old situation and the method is to complex for an overview at first glance.

Reply:
Is this better now?

Reply:
:) yes of course it is better, however...